seal.go

```diff
@@ -0,0 +1,370 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package secboot
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/chrisccoulson/go-tpm2"
+
+	"golang.org/x/xerrors"
+)
+
+func makeSealedKeyTemplate() *tpm2.Public {
```

Comment:
I would move this to keydata.go,  another approach (but maybe overkill) would be to combine my proposal and this:

in keydata.go:

```
var sealedKeySkeleton = tpm2.Public{
		Type:    tpm2.ObjectTypeKeyedHash,
		NameAlg: tpm2.HashAlgorithmSHA256,
		Attrs:   tpm2.AttrFixedTPM | tpm2.AttrFixedParent,
		Params:  tpm2.PublicParamsU{Data: &tpm2.KeyedHashParams{Scheme: 
                 tpm2.KeyedHashScheme{Scheme: tpm2.KeyedHashSchemeNull}}}}
}
```
leave out Params if they are mutable, sealedKeySkeleton.Xyz can be used in lots of places

here in seal.go:

```
func makeSealedKeyTemplate(authPolicy) *tmp.Public {
     t := sealedKeySkeleton // copy
     t.AuthPolicy = authPolicy
     return &t
}
```